(Feature) Beautify modals

gabitoesmiapodo opened the pull request:
This PR closes [#167223377](https://www.pivotaltracker.com/story/show/167223377), by:

* Adding quite a bunch of new styles and refactoring.
* Updating too many things to list.

**Extra:**

I don't know if this needs too much explanation or details.

- Modal's bottom jagged borders can be disabled dynamically.
- I revised some other styles along the way (feed items, dashboard content, etc.), because there were some related components I needed to modify.
- I created a generic modal wrapper which I might reuse when styling the dialogs next (not sure about this yet, but the base structure and code should be useful anyway).

You should be able to see something like this in the dashboard:

![Screen Shot 2019-07-12 at 16 34 32](https://user-images.githubusercontent.com/4015436/61154459-89906e00-a4c4-11e9-9454-d9ebae6c5991.png)

The modals should look something like this (Send / Claim / Receive modals are currently working):

![Screen Shot 2019-07-12 at 16 34 37](https://user-images.githubusercontent.com/4015436/61154520-b5abef00-a4c4-11e9-824d-99440f8edc74.png)
![Screen Shot 2019-07-12 at 16 34 44](https://user-images.githubusercontent.com/4015436/61154522-b5abef00-a4c4-11e9-874e-2c15ded835cc.png)
<img width="280" alt="Screen Shot 2019-07-12 at 16 34 59" src="https://user-images.githubusercontent.com/4015436/61154523-b6448580-a4c4-11e9-8547-a5eb295bb057.png">


<!-- **Considerations:** -->
<!-- * PR Title Convention: (Feature|Bug|Chore) Task Name -->
<!-- * Be as descriptive as possible, assisting reviewers as much as possible. -->
<!-- * If frontend, paste screenshots that display the UI changes. -->
<!-- * If there is interactivity, paste a gif showing the feature. -->

fernandomg commented on src/components/common/modal/ModalWrapper.js:
```diff
@@ -0,0 +1,186 @@
+      fontSize: normalize(20),
+      fontWeight: '700',
+    },
+    rightTitle: {
```
it seems that there are several styles definitions duplicated across components and not used (this one in particular is in: `ModalSendEvent`, `ModalReceivEvent`, `ModalWrapper`, `EventStyles`)

gabitoesmiapodo replied:
Deleted _ModalSendEvent_ and _ModalReceivEvent_, as they don't seem to be used anywhere.

Removed unused styles from _ModalWrapper_

Same for _EventStyles_

fernandomg commented on src/components/dashboard/Dashboard.js:
```diff
@@ -251,6 +247,17 @@ const styles = StyleSheet.create({
     height: '256px',
   },
+  bigNumberStyles: {
+    fontFamily: 'Roboto-Bold',
```
`RobotSlab-Bold` is stated in wireframes

gabitoesmiapodo replied:
Changed it on the dashboard's main big number.

fernandomg commented on src/components/dashboard/Dashboard.js:
```diff
@@ -251,6 +247,17 @@ const styles = StyleSheet.create({
+    marginRight: normalize(3),
+  },
+  bigNumberUnitStyles: {
+    fontFamily: 'Roboto-Bold',
```
`RobotSlab-Bold` is stated in wireframes

gabitoesmiapodo replied:
Changed it on the dashboard's main big number.

fernandomg commented on src/components/dashboard/FeedItems/FeedModalItem.js:
```diff
@@ -1,148 +1,151 @@
+        claim: require('./img/receive.png'),
+        receive: require('./img/receive.png'),
+        send: require('./img/send.png'),
+      }[type] || null
```
:ok_hand:

gabitoesmiapodo replied:
I obviously stole that pattern from you.

fernandomg commented on src/components/dashboard/FeedItems/ListEventItem.js:
```diff
@@ -60,38 +55,63 @@ const ListEvent = ({ item: feed, theme, styles }: FeedEventProps) => {
   },
   mainSection: {
-    marginLeft: normalize(4),
+    marginLeft: normalize(8),
```
shouldn't we use `theme.paddings` values here? There's a `padding` set for `innerRow` in this same object.

gabitoesmiapodo replied:
No. Global styles have nothing to do here.

It may be that we could use a better name for this style / component, but having a local style for this is correct IMO.

fernandomg commented on src/components/dashboard/FeedItems/ListEventItem.js:
```diff
@@ -60,38 +55,63 @@ const ListEvent = ({ item: feed, theme, styles }: FeedEventProps) => {
+    display: 'flex',
+    justifyContent: 'space-between',
+    marginBottom: normalize(8),
+    paddingBottom: normalize(5),
```
Same `theme.paddings` thing.

Particularly about `paddingBottom`, I'm not sure if it has to be `5` or a value multiple of `4`

gabitoesmiapodo replied:
Changed `5` to `4`.

fernandomg commented on src/components/dashboard/FeedItems/ListEventItem.js:
```diff
@@ -60,38 +55,63 @@ const ListEvent = ({ item: feed, theme, styles }: FeedEventProps) => {
-    color: '#A3A3A3',
-    marginTop: normalize(4),
+    marginTop: normalize(5),
+    paddingRight: normalize(5),
```
Same thing about values being multiple of `4`

gabitoesmiapodo replied:
Changed `5` to `4`.

sirpy commented:
@gabitoesmiapodo just want to make sure you accounted for all the different types of feed modals.
specifically for this multi button modal
https://xd.adobe.com/spec/fb6d4d7e-dd16-44e1-4bf3-1df4afd50d5c-8d5c/screen/6e285b2a-dd1d-478e-9e6c-2f3a5c0fde85/Dashboard-Feed-PopUp-Sent-GD-by-link-Pending-

gabitoesmiapodo commented:
@fernandomg 

> Text goes out of boundaries

Fixed.

![Screen Shot 2019-07-15 at 09 41 33](https://user-images.githubusercontent.com/4015436/61216917-e7030580-a6e4-11e9-82b2-6bcc8ea0e29e.png)

> zIndex is expecting number instead of a string. Browser is showing this error:
Warning: Failed prop type: Invalid prop `zIndex` of type `string` supplied to `View`, expected `number`.

Fixed.

> tests are failing. Need to update snapshots

Will do.

gabitoesmiapodo commented:
@sirpy 

Yes, I'm aware those designs exist but they're not implemented yet, so I won't be able to style them.
